### PR TITLE
DDF-2803 Re-add deprecated GzipFilter to web apps to improve element compression

### PR DESCRIPTION
**Pull request by mweser**
#### What does this PR do?
Re-adds deprecated GzipFilter to web app pom.xml files to improve element compression. Future task will involve fixing issue with Pax Web update not playing nicely with Jetty with the newer GzipHandler servlet.

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged) @ryeats @andrewkfiedler @djblue 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@stustison 
@lessarderic 

#### How should this be tested? (List steps with links to updated documentation)
1. Build and install DDF through Admin UI
2. Open "Inspect" to "Network" tab and refresh through different UI pages in Admin UI
    and Catalog
3. Verify that gzip is occurring (check "Content-Encoding")
4. Verify this throughout web apps

#### What are the relevant tickets?
[](https://codice.atlassian.net/projects/DDF/issues/DDF-2803)

#### Screenshots (note transfer size for elements vs decompressed size)
screen shot 2017-08-10 at 09 48 38

screen shot 2017-08-14 at 14 55 28

screen shot 2017-08-14 at 14 56 10

screen shot 2017-08-14 at 15 00 59

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/pom.xml [hunk 1]
```diff
@@ -204,7 +204,7 @@
         <dependency>
             <groupId>org.eclipse.jetty</groupId>
             <artifactId>jetty-servlets</artifactId>
-            <version>${jetty.version}</version>
+            <version>9.2.19.v20160908</version>
```
Comment (lessarderic) [inline catalog/ui/catalog-ui-search/pom.xml]: ❗️ We should put a comment here to explain why we're doing this.

**Comment by mweser** [comment]
Updated dependencies in 39 files so that all web apps are gzipping.

**Comment by mweser** [comment]
build now

**Comment by mweser** [comment]
Passes full swarm build.

File: broker/broker-undelivered-messages-ui/src/main/resources/WEB-INF/web.xml [hunk 2]
```diff
@@ -40,5 +40,20 @@
...
         </user-data-constraint>
     </security-constraint>
+
+    <filter>
+        <filter-name>GzipFilter</filter-name>
+        <filter-class>org.eclipse.jetty.servlets.GzipFilter</filter-class>
+        <init-param>
+            <param-name>mimeTypes</param-name>
+
+            <param-value>text/html,text/plain,text/xml,application/xhtml+xml,text/css,application/javascript,image/svg+xml</param-value>
```
Comment (djblue) [inline broker/broker-undelivered-messages-ui/src/main/resources/WEB-INF/web.xml]: Are these the only mime types that are run through gzip? Have you tried looking at `application/json`?

File: catalog/ui/catalog-ui-search/pom.xml (hunk 1 above)
Comment (mweser) [inline catalog/ui/catalog-ui-search/pom.xml]: @lessarderic Something along the lines of "Temporarily adding deprecated GzipFilter functionality until GzipHandler supported by PaxWeb"?

File: broker/broker-undelivered-messages-ui/src/main/resources/WEB-INF/web.xml (hunk 2 above)
Comment (mweser) [inline broker/broker-undelivered-messages-ui/src/main/resources/WEB-INF/web.xml]: This should be resolved in https://codice.atlassian.net/browse/DDF-3217

**Comment by mweser** [comment]
build now

**Comment by AzGoalie** [comment]
Manually verified that the webapps have gzip encoding.

**Comment by mweser** [comment]
Passes full swarm build.

**Comment by bdeining** [comment]
build now

**Pull request merged**